### ISSUE TITLE
renconstruct.exe fails with error "tables.nim(234Error: unhandled exception: key not found: manifest [KeyError]"

### ISSUE THREAD

**Issue by SergioAlonsoJr** [kobaltcore/renkit#3]
Hi, 

Your script is exactly what I'm looking for, but it fails and I don't know where is the error.

Below is a screenshot of my .toml and the command line execution.

![image](https://user-images.githubusercontent.com/4298411/151682412-3926ccae-9b20-4402-ac93-c9a74e4b7159.png)

I'm on Windows.

Can you help me?

**Comment by kobaltcore** [comment]
Ah, I do believe that this was fixed in my last few commits, since this is complaining about an optional key in the config file not being present.
Making my config look like yours and running a test build, mine passed without issues on the latest version.
Alas, I forgot to actually *publish* this as a new release, which I have now done. If you upgrade to this one, it should hopefully work. You can find the prebuilt binaries [here](https://github.com/kobaltcore/renkit/releases/tag/v1.1.0).

Let me know if you run into any other issues, I'll be happy to help.

**Comment by SergioAlonsoJr** [comment]
Thanks, @kobaltcore, the new version fixed that error!

I have a new problem now:

When I build using Ren'py interface, the game size is 1.8 GB (correct). When I build using renconstruct, the game is size only 31 MB and the game folder is almost empty.

In the script below you can see that the command ran to completion, that my project folder has a lot of content, and that the build version has no content at all.

![image](https://user-images.githubusercontent.com/4298411/151704375-cdf7fe20-9f70-43f5-98a3-a664f791e421.png)

Also, the name of my project is not showing in the .zip

I make use of the [build.classify](https://www.renpy.org/doc/html/build.html#classifying-and-ignoring-files) to select the specific files that go into the project.

**Comment by kobaltcore** [comment]
The name missing from the output is indeed quite strange and would indicate that something goes wrong while packaging the game. The thing here is that renconstruct simply hands off to Ren'Pys usual build process and doesn't really interfere with it at all, so that is some odd behavior.

Trying this on a sample project on my local PC I am actually able to reproduce this issue. Specifically this happens when the path to your game files is not correct, in which case Ren'Py will silently build an empty project (hence the missing name).
This is probably the case because your path (judging from your last screenshot) contains an URL-encoded space as `%20`, which is not a valid path.

I'm going to add a guard against this so this issue won't pop up in the future. For now, the best way to fix this is to ensure that the path you supply is actually valid and can be resolved. With Windows this can often be a pain since it's different from all the Unix-based OS's and unfortunately I'm on macOS myself, so I can't really support you in finding the exact path on your end.

Copying it from Windows Explorer and supplying it in quotes `-i "<my-path>"` might be the best option.

The new version I just released will tell you if the path is valid or not, so that can be a first smoke test. It's available [here](https://github.com/kobaltcore/renkit/releases/tag/v1.1.1).

Generally I would recommend against using spaces in your file paths to avoid issues like this, since spaces can often mess with CLI parsing.

**Comment by SergioAlonsoJr** [comment]
Hey, @kobaltcore,

It's a good thing you touched the subject of the dir.

I copied and pasted the folder directory from Windows Explorer and this is what I got

![image](https://user-images.githubusercontent.com/4298411/151706250-de254714-2b3c-498c-a37d-12105daa7f7e.png)

On the other hand, when I try to build the project directly in the ren'py folder with the code below, it works (but I want to use your solution because it's offer more features)

`renpy.exe launcher distribute "D:\VN Dropbox\Escrita\visual-novel-demo"`

**Comment by kobaltcore** [comment]
You may try supplying your path with the space escaped, i.e. `"D:\VN\ Dropbox\Escrita\..."` (note the extra `\` before the space).
In case this does not work, copying your game next to renconstruct (or vice versa, copying renconstruct next to your game folder) and then running it with a relative path would likely also work, i.e. `renconstruct.exe -i "./visual-novel-demo"`, where renconstruct is right next to your game folder.

**Comment by kobaltcore** [comment]
Hey, just a quick addendum to my last post: I have attempted to make `renkit` more robust when dealing with paths containing spaces, so maybe give `1.1.2` a try, which ships these new improvements. I just tried this on a game folder with spaces on it on my machine, and there it worked, so I'm hopeful that it will work for you as well.
The prebuilt binaries are available [here](https://github.com/kobaltcore/renkit/releases/tag/v1.1.2).
